micromamba/src/update.cpp

```diff
@@ -8,10 +8,121 @@
 
 #include "mamba/api/configuration.hpp"
 #include "mamba/api/update.hpp"
+#include "mamba/api/channel_loader.hpp"
 
+#include "mamba/core/transaction.hpp"
+#include "mamba/core/context.hpp"
+#include "mamba/core/util_os.hpp"
+#include "version.hpp"
+
+extern "C"
+{
+#include "solv/evr.h"
+#include "solv/pool.h"
+#include "solv/conda.h"
+#include "solv/repo.h"
+#include "solv/selection.h"
+#include "solv/solver.h"
+}
 
 using namespace mamba;  // NOLINT(build/namespaces)
 
+int
+update_self(const std::optional<std::string>& version)
+{
+    auto& config = mamba::Configuration::instance();
+    auto& ctx = mamba::Context::instance();
+    config.load();
+
+    mamba::MPool pool;
+    mamba::MultiPackageCache package_caches(ctx.pkgs_dirs);
+
+    auto exp_loaded = load_channels(pool, package_caches, 0);
+    if (!exp_loaded)
+    {
+        throw std::runtime_error(exp_loaded.error().what());
+    }
+
+    pool.create_whatprovides();
+    std::string matchspec;
+    if (!version)
+    {
+        matchspec = fmt::format("micromamba>{}", umamba::version());
```

Comment:
Would this work? It would avoid unnecessary duplications.
```
const auto version_to_use = version ? version.value() : umamba::version();
matchspec = fmt::format("micromamba>{}", version);
```

Reply:
It would not because one is `=` and the other `>` but yeah, it could be slightly shorter although I think the gain is not that big :)